app/controllers/compliments_controller.rb

```diff
@@ -1,11 +1,15 @@
 class ComplimentsController < ApplicationController
 
   def index
-    @compliment = Compliment.order("random()").limit(1).first
+    @compliment = Compliment.random
   end
 
   def show
-    @compliment = Compliment.find(params[:id])
+    @compliment = if params[:id] == 'random'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.